typing/typecore.ml

```diff
@@ -4437,14 +4492,14 @@ and type_application env (mode : Alloc_mode.t) funct sargs =
   match sargs with
   | (* Special case for ignore: avoid discarding warning *)
     [Nolabel, sarg] when is_ignore funct ->
-      let ty_arg, ty_res = filter_arrow env (instance funct.exp_type) Nolabel in
-      (* FIXME arg mode *)
-      let exp = type_expect env mode sarg (mk_expected ty_arg) in
+      let marg, ty_arg, _mres, ty_res =
```

Comment:
Should probably check that the resulting mode matches the mode of the application, in case someone binds `%ignore` as returning a local.